treemap/test/treemap-test-pptr.js

```diff
@@ -101,53 +99,48 @@ describe('Lighthouse Treemap', () => {
       });
     }
 
-    it('from window postMessage', async () => {
-      await loadFromPostMessage(debugOptions);
-      const optionsInPage = await page.evaluate(() => window.__treemapOptions);
-      expect(optionsInPage.lhr.requestedUrl).toBe(debugOptions.lhr.requestedUrl);
-    });
-
-    it('handles errors', async () => {
-      await loadFromPostMessage({});
-      const optionsInPage = await page.evaluate(() => window.__treemapOptions);
-      expect(optionsInPage).toBeUndefined();
-      const error = await page.evaluate(() => document.querySelector('#lh-log').textContent);
-      expect(error).toBe('Error: Invalid options');
-    });
-
-    it('from encoded fragment (gzip)', async () => {
+    it('from encoded fragment (no gzip)', async () => {
       const options = JSON.parse(JSON.stringify(debugOptions));
       options.lhr.requestedUrl += '😃😃😃';
-      const json = JSON.stringify(options);
-      const encoded = await page.evaluate(`
-        ${getTextEncodingCode()}
-        TextEncoding.toBase64(${JSON.stringify(json)}, {gzip: true});
-      `);
-
-      await page.goto(`${treemapUrl}?gzip=1#${encoded}`);
-      await page.waitForFunction(
-        () => window.__treemapOptions || document.body.textContent.startsWith('Error'));
+      await loadFromEncodedUrl({options, usesGzip: false});
 
       const optionsInPage = await page.evaluate(() => window.__treemapOptions);
       expect(optionsInPage.lhr.requestedUrl).toBe(options.lhr.requestedUrl);
     });
 
-    it('from encoded fragment (no gzip)', async () => {
+    it('from encoded fragment (gzip)', async () => {
       const options = JSON.parse(JSON.stringify(debugOptions));
       options.lhr.requestedUrl += '😃😃😃';
-      const json = JSON.stringify(options);
-      const encoded = await page.evaluate(`
-        ${getTextEncodingCode()}
-        TextEncoding.toBase64(${JSON.stringify(json)}, {gzip: false});
-      `);
-
-      await page.goto(`${treemapUrl}#${encoded}`);
-      await page.waitForFunction(
-        () => window.__treemapOptions || document.body.textContent.startsWith('Error'));
+      await loadFromEncodedUrl({options, usesGzip: true});
 
       const optionsInPage = await page.evaluate(() => window.__treemapOptions);
       expect(optionsInPage.lhr.requestedUrl).toBe(options.lhr.requestedUrl);
     });
+
+    const errorTestCases = [
+      {
+        options: {lhr: 'lol'},
+        error: 'Error: provided json is not a Lighthouse result',
+      },
+      {
+        options: {lhr: {noaudits: {}}},
+        error: 'Error: provided json is not a Lighthouse result',
+      },
+      {
+        options: {lhr: {audits: {}}},
+        error: 'Error: provided Lighthouse result is missing audit: `script-treemap-data`',
+      },
+    ];
+    for (let i = 0; i < errorTestCases.length; i++) {
+      it(`handles errors #${i + 1}`, async () => {
```

Comment:
Can you give these their own `describe` block? It's hard to target specific tests in vscode when there is no plaintext name to target.